### PR TITLE
PR 199: Implement a snapshot creating function for vms

### PR DESCRIPTION
**Pull request by szimszon**
Implemented a function called snapshot_create in vm.py. So now we are able to create 
snapshots with this library.Also there is a small example script called 
make-vm-snapshot.py to demonstrate this functionality, along with a unit test case.

### CODE REVIEW

**Comment by szimszon** [comment]
Hi @sompa ,

Sorry, what do you mean with "testing information for your changes"?

**Comment by hodgesrm** [comment]
Hi @szimszon, 

First of all thanks for the contribution!  We don't have snapshot manipulation yet so it's great you are willing to help us hack it in. 

I added a few specific comments about trivial fixes required to merge.  More generally to expand on the comment from @sompa we need new features to be tested.  At the minimum we need a unit test case that generates a snapshot.  Probably the simplest approach is to add a new test case to tests/vcd_vm.py.  

Cheers, Robert

File: pyvcloud/vcd/vm.py [hunk 1]
```diff
@@ -133,6 +133,25 @@ def power_reset(self):
         return self.client.post_linked_resource(
             self.resource, RelationType.POWER_RESET, None, None)
 
+    def snapshot_create(self, memory=None, quiesce=None, name=None):
+        """Snapshot the VM.
+
```
[the remark this answers is not in the document]
Comment (szimszon) [reply pyvcloud/vcd/vm.py]: Done.

File: examples/make-vm-snapshot.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+print("Fetching vApp...")
+vapp_resource = vdc.get_vapp(vapp)
+vapp = VApp(client, resource=vapp_resource)
+
+print("Fetching VM...")
+vm_resource = vapp.get_vm(vm)
+vm = VM(client, resource=vm_resource)
+
+print("Createing Snapshot...")
```
[the remark this answers is not in the document]
Comment (szimszon) [reply examples/make-vm-snapshot.py]: Replaced.

File: examples/make-vm-snapshot.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+# This product is licensed to you under the
+# Apache License, Version 2.0 (the "License").
+# You may not use this product except in compliance with the License.
+#
+# This product may include a number of subcomponents with
+# separate copyright notices and license terms. Your use of the source
+# code for the these subcomponents is subject to the terms and
+# conditions of the subcomponent's license, as noted in the LICENSE file.
+#
+# Illustrates how to list all vApps within a single vDC.
```
[the remark this answers is not in the document]
Comment (szimszon) [reply examples/make-vm-snapshot.py]: Changed.

**Comment by szimszon** [comment]
There is also a test case in pyvcloud/tests/vcd_vm.py for snapshot creation...

**Comment by hodgesrm** [comment]
Thanks @szimszon.  Your PR is merged.  We appreciate the contribution!

**Pull request merged**